### PR TITLE
Add tool to fetch all plugin asciidocs

### PR DESCRIPTION
**Pull request by jordansissel**
I planned to write this in Java or Groovy, but today was a day I had to wrestle IntelliJ, and I lost that battle.

Back to vim and ruby for this in order to move forward.

### CODE REVIEW

**Comment by jordansissel** [comment]
I will create a PR for demonstrating/evaluating what this will do in logstash-docs.

**Comment by jordansissel** [comment]
Demo of what this changes if run into logstash-docs: https://github.com/elastic/logstash-docs/compare/demo-plugins-sync?expand=1

**Comment by jordansissel** [comment]
```
% ruby ../docs-tools/plugindocs.rb --output-path /home/jls/projects/logstash-docs ../docs-tools/plugins_version_docs.json
```

File: plugindocs.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    plugins = report["successful"]
+
+    skip = [ "logstash-core-plugin-api", "logstash-patterns-core", "logstash-devutils" ]
+    skip += [ 
+      "logstash-codec-sflow", # Empty plugin repository
+      "logstash-filter-math", # Empty plugin repository
+      "logstash-input-mongodb", # Empty plugin repository
+      "logstash-filter-kubernetes_metadata", # https://github.com/logstash-plugins/logstash-filter-kubernetes_metadata/issues/2
+      "logstash-input-eventlog", # https://github.com/logstash-plugins/logstash-input-eventlog/issues/38
+    ]
```
Comment (ph) [inline plugindocs.rb]: Maybe just put the list in an external yml file?

File: plugindocs.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      File.write(output_asciidoc, response.body) 
+      puts repository
+    end
+  end
+end
+
+if __FILE__ == $0
+  PluginDocs.run
+end
```
Comment (ph) [inline plugindocs.rb]: +1 to use fetch the raw file directory from github instead of cloning.

**Comment by ph** [comment]
With your move to rbenv were you successful to generate the `plugins_version_docs.json` 
 from https://github.com/elastic/logstash/pull/6941/ ?

File: plugindocs.rb (hunk 1 above)
Comment (ph) [inline plugindocs.rb]: I've created https://github.com/logstash-plugins/logstash-input-eventlog/pull/39, once its merged we can remove this ignore.

**Comment by jordansissel** [comment]
Added `--master` flag for fetching all docs from plugins' master branch

**Comment by jordansissel** [comment]
Added replacement of %VERSION% and friends.

**Comment by jordansissel** [comment]
I added a yaml config file per @jsvd's suggestion. Currently only contains the repository skip list.

**Comment by jordansissel** [comment]
@ph can you do another review of this? I think it's ready.

**Comment by ph** [comment]
post merge LGTM, I didn't this ping sorry about that @jordansissel

**Pull request merged**